### PR TITLE
PBAC : information incident fixes

### PR DESCRIPTION
**Pull request by Tom-Ritense**
2 incidents are fixed in this pr.
https://ritense.tpondemand.com/entity/125821-be-check-for-execution-access-before
https://ritense.tpondemand.com/entity/125820-be-check-for-document-access-before

### CODE REVIEW

File: form/src/main/kotlin/com/ritense/form/service/PrefillFormService.kt [hunk 1]
```diff
@@ -86,7 +90,16 @@ class PrefillFormService(
         val formDefinition = formDefinitionService.getFormDefinitionById(formDefinitionId)
             .orElseThrow { RuntimeException("Form definition not found by id $formDefinitionId") }
         if (documentId != null) {
-            val document = runWithoutAuthorization { documentService.get(documentId.toString()) }
+            val document = runWithoutAuthorization {
```
Comment (Klaas-Ritense) [inline form/src/main/kotlin/com/ritense/form/service/PrefillFormService.kt]: I think you could just remove the `runWithoutAuthorization` here. Then you won't need the `authorizationService`

File: process-link/src/main/kotlin/com/ritense/processlink/service/ProcessLinkActivityService.kt [hunk 2]
```diff
@@ -64,19 +74,63 @@ class ProcessLinkActivityService(
         val processLink = processLinkService.getProcessLinksByProcessDefinitionIdAndActivityType(
             processDefinitionId,
             ActivityTypeWithEventName.START_EVENT_START
-        ) ?: return null
+        ) ?: throw ProcessLinkNotFoundException("For process definition with id $processDefinitionId")
```
Comment (Klaas-Ritense) [inline process-link/src/main/kotlin/com/ritense/processlink/service/ProcessLinkActivityService.kt]: I don't thing the FE knows how to react to this exception yet. The will no longer show the 'please configure a start form' -message

File: process-link/src/main/kotlin/com/ritense/processlink/service/ProcessLinkActivityService.kt [hunk 3]
```diff
@@ -64,19 +74,63 @@ class ProcessLinkActivityService(
...
+            processLinkActivityHandlers
+                .find { it.supports(processLink) }
+                ?.getStartEventObject(processDefinitionId, documentId, documentDefinitionName, processLink)
         }
     }
 
     companion object {
         val logger = KotlinLogging.logger {}
+
+        fun createDummyCamundaExecution(
```
Comment (Klaas-Ritense) [inline process-link/src/main/kotlin/com/ritense/processlink/service/ProcessLinkActivityService.kt:104]: hmm. If it works, it works. I guess

File: process-link/src/main/kotlin/com/ritense/processlink/service/ProcessLinkActivityService.kt [hunk 4]
```diff
@@ -64,19 +74,63 @@ class ProcessLinkActivityService(
         val processLink = processLinkService.getProcessLinksByProcessDefinitionIdAndActivityType(
             processDefinitionId,
             ActivityTypeWithEventName.START_EVENT_START
-        ) ?: return null
+        ) ?: throw ProcessLinkNotFoundException("For process definition with id $processDefinitionId")
+
+        val processDefinition = runWithoutAuthorization {
```
Comment (Tom-Ritense) [inline process-link/src/main/kotlin/com/ritense/processlink/service/ProcessLinkActivityService.kt:79]: @Klaas-Ritense not sure if this is right place.

File: form/src/main/kotlin/com/ritense/form/service/PrefillFormService.kt [hunk 5]
```diff
@@ -58,7 +58,8 @@ class PrefillFormService(
     private val formFieldDataResolvers: List<FormFieldDataResolver>,
     private val processDocumentAssociationService: ProcessDocumentAssociationService,
     private val valueResolverService: ValueResolverService,
-    private val objectMapper: ObjectMapper
+    private val objectMapper: ObjectMapper,
+    private val authorizationService: AuthorizationService,
```
Comment (Klaas-Ritense) [inline form/src/main/kotlin/com/ritense/form/service/PrefillFormService.kt:62]: The AuthorizationService is no longer needed

**Review by Klaas-Ritense — APPROVED** [review]

**Pull request merged**